src/twisted/web/test/test_cgi.py

```diff
@@ -485,6 +486,20 @@ def test_prematureEndOfHeaders(self):
         self.assertEqual(request.responseCode, INTERNAL_SERVER_ERROR)
 
 
+    def test_connectionLost(self):
+        """
+        Ensure that the CGI process ends cleanly when the request connection
+        is lost.
+        """
+        d = DummyChannel()
+        request = http.Request(d, True)
+        protocol = twcgi.CGIProcessProtocol(request)
+        request.connectionLost(
+            failure.Failure(ConnectionLost("Connection done"))
+        )
+        protocol.processEnded(failure.Failure(error.ProcessTerminated()))
```

Comment:
Ideally tests should always have at least one assertion; I realize that this is just a regression test for something that previously caused an exception, so I am not going to be a stickler about it, but it would ideally verify *something* about the behavior of `processEnded`.